spring-integration-zeromq/src/main/java/org/springframework/integration/zeromq/ZeroMqUtils.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.zeromq;
+
+import org.zeromq.ZMQ;
+
+/**
+ * Module that wraps common methods of ZeroMq integration classes
+ *
+ * @author Alessio Matricardi
+ *
+ * @since 6.4
+ */
+public final class ZeroMqUtils {
+
+	/**
+	 * Binds the ZeroMq socket to the given port when different from 0
+	 * or to a random port if not.
+	 * @param socket the ZeroMq socket
+	 * @param port the port to bind ZeroMq socket to over TCP.
+	 * @return the effectively bound port
+	 * @since 6.4
+	 */
+	public static int bindSocket(ZMQ.Socket socket, int port) {
+		if (port == 0) {
+			return socket.bindToRandomPort("tcp://*");
```

Comment:
I wonder if it always has to be `tcp` protocol: http://api.zeromq.org/4-2:zmq-bind.

Or at least, let's document that in this method Javadoc!

Reply:
Fully agree with you.
I only merged the behaviour between `ZeroMqMessageProducer` and `ZeroMqMessageHandler`.

I believe that the support of the other transport protocols is out of the scope of this PR, if you don't mind I will make a specific one for this purpose.
Additionally, some protocols like `inproc` don't need a port but only an arbitrary string endpoint, so some rework is needed.

BTW, I will remark the fact that bind is actually possible only over TCP.

Reply:
Yeah... Let's look into that when concern arises!